### ISSUE TITLE
Assign None as default for Optional Type

### ISSUE THREAD

**Issue by dmfigol** [python-attrs/attrs#460]
Hi,
I think it would be awesome if the default value for Optional types would be set to None.
Here is a snippet I hoped would work:
```
import attr

@attr.s(auto_attribs=True)
class Book:
    name: str
    author: Optional[str]


data = {'name': 'Fluent Python'}
Book(**data)
```
```
TypeError: __init__() missing 1 required positional argument: 'author'
```

**Comment by euresti** [comment]
You can very easily add `= None` to the class definition.
Also default args don't play nice if they are in the middle.  e.g.

```
@attr.s(auto_attribs=True)
class Book:
    name: str
    author: Optional[str]
    isbn: str
```

**Comment by dmfigol** [comment]
This is what I am currently doing - assigning None to every field in 20-field schema and it is looking ugly.
I believe this is extra information which is already indicated by the Optional type.

**Comment by wsanchez** [comment]
I'm not sure I agree that the optional type indicates that the default should be `None`; it only indicates that the value may be `None`.

That said, it does seem a reasonable default-default.  But I'm a little wary of doing that implicitly, and @euresti's point about defaults in the middle would have to be addressed.

**Comment by hynek** [comment]
I don't think this would be culturally sensible anymore. I think the explicitness has proven to be an upside in general and behaving differently here would cause confusion elsewhere.

It _has_ been possible to achieve that oneself for a while using field transformers, tho: https://www.attrs.org/en/stable/extending.html#automatic-field-transformation-and-modification

Not sure it would be useful since it's impossible to type-check, tho.
